apps/passport-server/src/services/poapService.ts

```diff
@@ -104,18 +107,21 @@ export class PoapService {
     const pcd = await ZKEdDSAEventTicketPCDPackage.deserialize(parsed.pcd);
 
     logger(
-      `[POAP] checking that signer of ticket ${pcd.claim.partialTicket.ticketId} is passport-server`
+      `[POAP] checking that signer of ticket ${pcd.claim.partialTicket.ticketId} matches intended signer`
     );
-    if (!process.env.SERVER_EDDSA_PRIVATE_KEY)
-      throw new Error(`missing server eddsa private key .env value`);
 
-    const TICKETING_PUBKEY = await getEdDSAPublicKey(
-      process.env.SERVER_EDDSA_PRIVATE_KEY
-    );
+    if (signerPublicKey == null) {
+      if (!process.env.SERVER_EDDSA_PRIVATE_KEY)
+        throw new Error(`missing server eddsa private key .env value`);
+
+      signerPublicKey = await getEdDSAPublicKey(
+        process.env.SERVER_EDDSA_PRIVATE_KEY
+      );
+    }
 
     const signerMatch =
-      pcd.claim.signer[0] === TICKETING_PUBKEY[0] &&
-      pcd.claim.signer[1] === TICKETING_PUBKEY[1];
+      pcd.claim.signer[0] === signerPublicKey[0] &&
+      pcd.claim.signer[1] === signerPublicKey[1];
```

Comment:
There's an `isEqualEdDSAPublicKey` function in the `eddsa-pcd` package for doing these comparisons. It does the same thing, but perhaps less risk of a typo.